From f412cf0d214f823808797411cec10219b3fdddab Mon Sep 17 00:00:00 2001
From: Zhang Jiejing <jiejing.zhang@freescale.com>
Date: Thu, 28 Jun 2012 15:22:33 +0800
Subject: [PATCH 016/310] ENGR00215201 mx6: add serial no for user build, add
 jni opts.

- add serial no for user build, this serial no is known by
  most of host application.
- add jni opts for don't crash on jnicheck on eng build,
  some application maybe have issue on JNI call, and some
  customer release theri product as eng build, this maybe
  cause some application not run, like "swing shot".

Signed-off-by Zhang Jiejing <jiejing.zhang@freescale.com>
---
 imx6/etc/init.rc | 11 ++++++++---
 1 file changed, 8 insertions(+), 3 deletions(-)

diff --git a/imx6/etc/init.rc b/imx6/etc/init.rc
index 837109b..7188a89 100644
--- a/imx6/etc/init.rc
+++ b/imx6/etc/init.rc
@@ -340,6 +340,11 @@ on boot
 # Enable USB host1'remote wakeup event wakeup android system
     write /sys/bus/platform/drivers/fsl-ehci/fsl-ehci.1/power/wakeup disabled
 
+# serial no, vendor id,etc configure
+    write /sys/class/android_usb/android0/iManufacturer $ro.product.manufacturer
+    write /sys/class/android_usb/android0/iProduct $ro.product.model
+    write /sys/class/android_usb/android0/iSerial $ro.serialno
+
 # mount the debugfs
     mount debugfs none /sys/kernel/debug/
 
@@ -350,6 +355,9 @@ on boot
 # Set this property so surfaceflinger is not started by system_init
     setprop system_init.startsurfaceflinger 0
 
+# set this to not let jni check crash, just warnning message.
+    setprop dalvik.vm.jniopts warnonly
+
     class_start core
     class_start main
 
@@ -451,9 +459,6 @@ on property:ro.kernel.qemu=1
 # Always start adbd on userdebug and eng builds
 on property:ro.debuggable=1
     start console
-    write /sys/class/android_usb/android0/iManufacturer $ro.product.manufacturer
-    write /sys/class/android_usb/android0/iProduct $ro.product.model
-    write /sys/class/android_usb/android0/iSerial $ro.serialno
     start adbd
 
 # This property trigger has added to imitiate the previous behavior of "adb root".
-- 
1.8.0

